src/engine/enginemaster.cpp

```diff
@@ -446,15 +515,24 @@ void EngineMaster::process(const int iBufferSize) {
         // Perform balancing on main out
         SampleUtil::applyAlternatingGain(m_pMaster, balleft, balright, iBufferSize);
 
-        // Update VU meter (it does not return anything). Needs to be here so that
-        // master balance is reflected in the VU meter.
-        if (m_pVumeter != NULL) {
-            m_pVumeter->process(m_pMaster, iBufferSize);
-        }
         // Submit master samples to the side chain to do shoutcasting, recording,
         // etc. (cpu intensive non-realtime tasks)
+        CSAMPLE* pSidechain = m_pMaster;
         if (m_pSideChain != NULL) {
-            m_pSideChain->writeSamples(m_pMaster, iBufferSize);
+            if (m_pMasterTalkoverMix->toBool()) {
+                // Add Talkover to Sidechain output, re-use the talkover buffer
+                SampleUtil::addWithGain(m_pTalkover,
+                        m_pMaster, 1.0,
+                        iBufferSize);
+                pSidechain = m_pTalkover;
+            }
+            m_pSideChain->writeSamples(pSidechain, iBufferSize);
+        }
+
+        // Update VU meter (it does not return anything). Needs to be here so that
+        // master balance and talkover is reflected in the VU meter.
+        if (m_pVumeter != NULL) {
```

Comment:
Yea, slightly sloppy but It doesn't seem worth it to add a new sidechain VU meter.